src/Compilers/VisualBasic/Portable/Lowering/StateMachineRewriter/StateMachineStates.vb

```diff
@@ -10,9 +10,9 @@ Imports Microsoft.CodeAnalysis.VisualBasic.Syntax
 Namespace Microsoft.CodeAnalysis.VisualBasic
 
     Friend Module StateMachineStates
```

Comment:
>Friend Module StateMachineStates [](start = 4, length = 32)

I think that at this point we can make this an Enum. Is there a reason not to?

Reply:
@AlekseyTs I did try an implement and it worked, compiled and passed the unittest on my machine. The proposed implementation is in a reply above.

Reply:
All issue is if there are any external use _i can't observe_ that could be affected by the change of type / representation.

Reply:
It's an internal type in an assembly that isn't available for IVT.  So you should be fine to make this change.  thanks :)

Reply:
> Only issue is if there are any external use i can't observe that could be affected by the change of type / representation.

We generally think that we are free to change internal/friend declaration as we wish. 

---
In reply to: [400488739](https://github.com/dotnet/roslyn/pull/42870#discussion_r400488739) [](ancestors = 400488739)